api/v1alpha1/authorization_types.go

```diff
@@ -72,6 +76,18 @@ type Principal struct {
 	// JWT authentication in the same `SecurityPolicy`.
 	// +optional
 	JWT *JWTPrincipal `json:"jwt,omitempty"`
+
+	// Headers authorize the request based on the headers in the request.
+	// If multiple headers are specified, all headers must match for the rule to match.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	Headers []HeaderMatch `json:"headers,omitempty"`
+
+	// Methods authorize the request based on the HTTP methods.
+	// If multiple methods are specified, one of the methods must match for the rule to match.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	Methods []gwapiv1.HTTPMethod `json:"methods,omitempty"`
```

Comment:
Methods could also be support as pseudo headers(:method) within the `Headers` field, however, providing a separate field would be more user-friendly and improve UI clarity.